spec/services/products_renderer_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'spec_helper'
+
+describe ProductsRenderer do
+  let(:distributor) { create(:distributor_enterprise) }
+  let(:order_cycle) { create(:simple_order_cycle, distributors: [distributor]) }
+  let(:exchange) { order_cycle.exchanges.to_enterprises(distributor).outgoing.first }
+  let(:customer) { create(:customer) }
+  let(:pr) { ProductsRenderer.new(distributor, order_cycle, customer) }
```

Comment:
Can we rename it to `products_renderer`? IMO acronyms like this hurt readability